packages/auth/src/grant/routes.test.ts

```diff
@@ -115,162 +113,443 @@ describe('Grant Routes', (): void => {
-
-      await expect(grantRoutes.create(ctx)).resolves.toBeUndefined()
-      expect(ctx.response).toSatisfyApiSpec()
-      expect(ctx.status).toBe(200)
-      expect(ctx.body).toEqual({
-        interact: {
-          redirect: expect.any(String),
-          finish: expect.any(String)
-        },
-        continue: {
+        const ctx = createContext(
+          {
+            headers: {
+              Accept: 'application/json',
+              'Content-Type': 'application/json'
+            },
+            url,
+            method
+          },
+          {}
+        )
+        const body = {
           access_token: {
-            value: expect.any(String)
+            access: [
+              {
+                type: AccessType.IncomingPayment,
+                actions: [Action.Create, Action.Read, Action.List],
+                identifier: `https://example.com/${v4()}`
+              }
+            ]
           },
-          uri: expect.any(String),
-          wait: Config.waitTimeSeconds
+          client: CLIENT
         }
+        ctx.request.body = body
+
+        await expect(grantRoutes.create(ctx)).rejects.toHaveProperty(
+          'status',
+          500
+        )
       })
+      test('Fails to initiate a grant w/o interact field', async (): Promise<void> => {
+        const ctx = createContext(
+          {
+            headers: {
+              Accept: 'application/json',
+              'Content-Type': 'application/json'
+            },
+            url,
+            method
+          },
+          {}
+        )
 
-      scope.done()
+        ctx.request.body = { ...BASE_GRANT_REQUEST, interact: undefined }
+
+        await expect(grantRoutes.create(ctx)).rejects.toMatchObject({
+          status: 400,
+          error: 'interaction_required'
+        })
+      })
     })
 
-    test('Can get a software-only authorization grant', async (): Promise<void> => {
-      const ctx = createContext(
-        {
-          headers: {
-            Accept: 'application/json',
-            'Content-Type': 'application/json'
-          },
-          url,
-          method
-        },
-        {}
-      )
-      const body = {
-        access_token: {
-          access: [
-            {
-              type: AccessType.IncomingPayment,
-              actions: [Action.Create, Action.Read, Action.List],
-              identifier: `https://example.com/${v4()}`
-            }
-          ]
-        },
-        client: CLIENT
-      }
-      ctx.request.body = body
-
-      await expect(grantRoutes.create(ctx)).resolves.toBeUndefined()
-      expect(ctx.response).toSatisfyApiSpec()
-      expect(ctx.status).toBe(200)
-      expect(ctx.body).toEqual({
-        access_token: {
-          value: expect.any(String),
-          manage: expect.any(String),
-          access: body.access_token.access,
-          expires_in: 600
-        },
-        continue: {
+    describe('/continue', (): void => {
+      beforeEach(async (): Promise<void> => {
```

Comment:
```suggestion
      beforeAll(async (): Promise<void> => {
```

Reply:
I think it is actually not necessary here because I have it in line 117.